### ISSUE TITLE
signInWithWebUI hangs on dismissing the browser login on Android

### ISSUE THREAD

**Issue by gianmarcocalbi** [aws-amplify/amplify-flutter#1013]
**Describe the bug**
I have some logic in my app that is supposed to be executed after `signInWithWebUI` regardless the result is ok or an exception is thrown.. like this:

```dart
try {
  final result = await Amplify.Auth.signInWithWebUI(provider: provider);
  // standard logic
} catch (e, s) {
  // exception logic
}
```

Only on android, if the user dismisses the signin browser tab by tapping the X, neither the "standard logic" nor the "Exception logic" are executed, meaning the execution is somehow stuck at signInWithWebUI.
_I am using google as only provider._

**Expected behavior**
In the console, I get this message `D/AuthClient( 6412): Authorization flow canceled by user`, I am quite sure when I was on version 0.2.0 of the plugin I was also getting a `UserCancelledException` in flutter, but now nothing.

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
* [x] Android
* [ ] iOS

<details>

  <summary>Output of <code>flutter doctor -v</code></summary>
  
```
[✓] Flutter (Channel stable, 2.5.3, on macOS 12.0.1 21A559 darwin-arm, locale en-US)
    • Flutter version 2.5.3 at /Users/gcalbi/dev/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (2 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/gcalbi/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Users/gcalbi/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/gcalbi/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/gcalbi/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7784292/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Users/gcalbi/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/203.7621141/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.2)
    • IntelliJ at /Users/gcalbi/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Community Edition (version 2021.2)
    • IntelliJ at /Users/gcalbi/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/212.4746.92/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Community Edition (version 2021.1.3)
    • IntelliJ at /Users/gcalbi/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/211.7628.21/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2021.2)
    • IntelliJ at /Users/gcalbi/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/212.4746.92/IntelliJ IDEA.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.4746.57

[✓] IntelliJ IDEA Ultimate Edition (version 2021.1.3)
    • IntelliJ at /Users/gcalbi/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/211.7628.21/IntelliJ IDEA.app
    • Flutter plugin version 58.0.3
    • Dart plugin version 211.7727

[✓] VS Code (version 1.61.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 95.0.4638.54

• No issues found!

```

</details>

**Comment by haverchuck** [comment]
@gianmarcocalbi We have reproduced this and are testing a fix!

**Comment by dnys1** [comment]
Hi @gianmarcocalbi - we have released a fix for this issue as part of version 0.2.8. Would you be able to re-test on that version and let us know if you experience the same issue? Thanks!

**Comment by gianmarcocalbi** [comment]
> Hi @gianmarcocalbi - we have released a fix for this issue as part of version 0.2.8. Would you be able to re-test on that version and let us know if you experience the same issue? Thanks!

Hello!
Sure, I will let you know soon. Thanks

**Comment by gianmarcocalbi** [comment]
Hi @dnys1,
0.2.8 solves the issue, thanks!
